ext/spl/spl_observer.c

```diff
@@ -44,13 +44,19 @@ PHPAPI zend_class_entry     *spl_ce_MultipleIterator;
 
 PHPAPI zend_object_handlers spl_handler_SplObjectStorage;
 
+typedef struct _spl_SplObjectStorage_fptrs {
+	zend_function    *get_hash;
+	bool              override_read_dimension;
+	bool              override_write_dimension;
+} spl_SplObjectStorage_fptrs;
```

Comment:
Why is this a separate allocated struct instead of adding flags?

Reply:
I'd started out planning on making this an array of functions but decided to just track a boolean of whether the default object handlers should be used instead. I forgot about this while working on finishing the rest of the implementation - I agree this no longer makes sense.